lms/djangoapps/program_enrollments/api/writing.py

```diff
@@ -424,3 +449,52 @@ def _organize_requests_by_external_key(enrollment_requests):
             continue
         requests_by_key[key] = request
     return requests_by_key, duplicated_keys
+
+
+def _process_conflicted_course_enrollments(
```

Comment:
This function will actually delete keys from `requests_by_key`, can you change the name to `_remove_conflicted_course_enrollments`, so it's clear that it mutates its arguments?

Reply:
Alternatively, can you not modify `requests_by_key`, and instead return a new dict? Passing dictionaries back and forth while mutating them is a great way to make spaghetti code. It made the original APIView-based version of this API was very difficult to reason about.